internal/pkg/bully/server/bully.go

```diff
@@ -0,0 +1,239 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package bully
+
+import (
+	"context"
+	"log"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"github.com/talos-systems/talos/internal/pkg/bully/proto"
+	"google.golang.org/grpc"
+)
+
+// Address represents the address of a bully.
+type Address = string
+
+// Pid represents a process pid.
+type Pid = uint32
+
+// Processes represents the set of processes that form the cluster.
+type Processes struct {
+	*sync.RWMutex
```

Comment:
usually you can avoid dynamic allocation unless you want to pass `Processes` by value (looks like it's pointer in `Bully` struct). Zero value is good for `sync.RWMutex`, so you can drop the pointer type and just embed it as a value